### ISSUE TITLE
boolean heater instead of PWM?

### ISSUE THREAD

**Issue by flatsiedatsie** [r-downing/AutoPID#2]
Can this library drive a boolean heater as well?

**Comment by dtmutsvanga** [comment]
Use the  AutoPIDRelay class

**Comment by Grutsch** [comment]
Is their a sketch example for the AuoPIDRelay function

**Comment by dtmutsvanga** [comment]
No. The trick is to use a reasonably large value for your "pulse with." Try this:

    //Define Variables we'll be connecting to
    double Setpoint, Input, Output, temperature;
    double Kp=10, Ki=5, Kd=1;
    bool relayState;

    AutoPIDRelay myPID(&Input, &Setpoint, &relayState,10000.00,  Kp, Ki, Kd);
    // some code

    void loop() {
    temperature =getTempC();  //refresh temp
    myPID.run();
    digitalWrite(RELAY_PIN, !relayState);
    //do something
    }

**Comment by dtmutsvanga** [comment]
`    //Define Variables we'll be connecting to
    double Setpoint, Input, Output, temperature;
    double Kp=10, Ki=5, Kd=1;
    bool relayState;

    AutoPIDRelay myPID(&Input, &Setpoint, &relayState,10000.00,  Kp, Ki, Kd);
    // some code

    void loop() {
    temperature =getTempC();  //refresh temp
    myPID.run();
    digitalWrite(RELAY_PIN, !relayState);
    //do something
    }`

**Comment by Grutsch** [comment]
Thank's for the quick answer. I will try it straight away.
Alex

**Comment by Hollako** [comment]
Hello,

I am trying the AUTOPID Relay and i have a question please if the Pulsewidth value should be fixed in the sketch or it can be changed somewhere else inside the program like the Setpoint for example.

Thank you,

**Comment by dtmutsvanga** [comment]
AutoPIDRelay constructor takes pulseWidth as a parameter.
